src/framework/handlers/script.js

```diff
@@ -120,6 +145,38 @@ class ScriptHandler {
 
         head.appendChild(element);
     }
+
+    _loadModule(url, callback) {
+
+        // if we're in the browser, we need to use the full URL
+        const baseUrl = platform.browser ? window.location.origin : import.meta.url;
+        const importUrl = new URL(url, baseUrl);
+
+        // @ts-ignore
+        import(importUrl.toString()).then((module) => {
```

Comment:
So this uses import, which based on https://caniuse.com/?search=import is supported by Chrome from 2017, Safari 2018, Firefox 2019 - I wonder how we could handle the compatibility issues here.

Reply:
There has to be an assumption that if a user chooses to use ES modules, then there is a neccessary baseline level of support required. [ES6 is around 97.2%](https://caniuse.com/es6-module) which is only [0.7% less than our legacy build](https://browsersl.ist/#q=%3E+0.1%25%2C+partially+supports+webgl%2C+not+dead).

If someone uses the old script mechanism, the ESM path won't get triggered so it will just work as is.